test/smoke/packages/cert-manager/1.5.3/cert-manager-test.sh

```diff
@@ -73,17 +83,19 @@ spec:
     - www.example.com
 EOF
 
-sleep 45s
 
-status="$(kubectl get certificate/example-com | tail -n 1 | awk '{print $2}')"
-echo "${status}"
+echo "Waiting for certificate/example-com to Ready..."
+kubectl wait --for=condition=Ready certificate/example-com -n ${NAMESPACE} --timeout=300s
+status="$(kubectl get certificate/example-com -n ${NAMESPACE} -o 'jsonpath={..status.conditions[?(@.type=="Ready")].status}')"
```

Comment:
If `kubectl wait` fails, it means that the certificate is not ready, so it will stop the script execution as we have `set -o errexit`. If it passes, then it means that the certificate is ready and `status` will always be `True`.

So how our script will work is - if wait fails, then script ends with failure in pipeline but with no log like `cert-manager failed`. If wait passes, then script ends with success in pipeline with a log like `cert-manager Passed`

We could do something like this if we want to get around this

```bash
kubectl wait --for=condition=Ready certificate/example-com -n ${NAMESPACE} --timeout=300s || {
  echo "Failed waiting for certificate status to be ready"
  tanzu package installed delete cert-manager -y
  kubectl delete ns ${NAMESPACE}
  printf '\E[31m'; echo "cert-manager failed"; printf '\E[0m'
}

tanzu package installed delete cert-manager -y
kubectl delete ns ${NAMESPACE}
printf '\E[32m'; echo "cert-manager Passed"; printf '\E[0m'
```

We can also pull delete process into a function -

```bash
function cleanup {
  tanzu package installed delete cert-manager -y
  # but ${NAMESPACE} has to be set and available here
  kubectl delete ns ${NAMESPACE}
}
```

Reply:
The only small downside of the current script is, when wait fails, the cleanup misses to happen but the pipeline will fail if it runs the script. And cleanup is not so important in pipelines that are ephemeral - where the cluster is docker and the pipeline environment is a container or a throw away VM like in our case for most providers (AWS, Docker, Azure). Also, the cluster deployment tests will take care of deleting and cleaning up the whole cluster however. It's just that in local, if someone is using the tests independently and deletes the cluster later etc, it might be left lying around. Other than that, not much issues

We actually gotta fix many tricky cleanup issues that we missed all over the E2E scripts